text/orderer-v3.md

```diff
@@ -0,0 +1,320 @@
+- **Raft**: Several ordering nodes run the [Raft](https://raft.github.io/) protocol by embedding the [Raft library of etcd](https://github.com/etcd-io/etcd/tree/main/raft).
+
+Apart from the official Hyperledger Fabric consensus implementation, there is also a [Fabric fork](https://github.com/SmartBFT-Go/fabric) of a [BFT library](https://github.com/SmartBFT-Go/consensus) called SmartBFT.
+
+A BFT consensus protocol might have a different approach for onboarding and state snapshot replication compared to a CFT consensus protocol.
+In particular, a state snapshot sent between nodes in CFT is always assumed to be trusted, while in BFT it is not the case. 
+While it is tempting to have common code that performs state snapshot transfer that caters to both adversarial models, 
+such code is most likely to be complex and hard to follow and test.
+
+Each orderer type requires Fabric to understand how to initialize instances of it from the consensus metadata in the configuration blocks.
+However, this makes it impossible for external contributors to use their own consensus implementation without either forking Fabric entirely, 
+or making intrusive changes in the official Fabric orderer and peer code base.
+
+Undoubtedly, an alternative architecture which doesn't require Fabric codebase changes to add support for new 
+consensus protocols would be a sought-after property.
+
+# Identifying building blocks
+[building-blocks]: #identifying-building-blocks
+
+Apart from consensus, which agrees on orders of transaction, the orderer contains infrastructure that manages various aspects of its lifecycle and operations:
+
+ - **Chain management**: Spawns instances of the consensus protocols per channel and dispatches messages and operations to the right 
+   consensus instance.
+ - **Block replication service**: Server-side API that sends blocks upon request. 
+ - **Block synchronization utility**: Pulls blocks from other ordering service nodes and appends them to the local ledger. 
+ - **Communication service**: Authenticates client connections, receives messages from remote nodes and sends message to remote nodes.
+ - **Ledger subsystem**: Appends blocks to persistent storage and retrieves blocks by their index.
+ - **Channel participation service**: Manages local addition and removal of ledgers per channel.
+ - **Transaction filtering**: Enforces access control and validity checks on transactions both received from clients or from nodes.
+ - **Membership Service Provider subsystem**: Classifies identities into principals and drives policy authorization. 
+
+In this RFC, it is proposed that these building blocks be made consensus independent, and the building blocks that cannot be consensus independent will be 
+part of the consensus plugin that wraps around the consensus independent building blocks. 
+
+Therefore, the new organization of the components in the new ordering service node of a given type will look as follows:
+
```

Comment:
Can you be more explicit about an "ordering service node of a given type". That is, highlight that an ordering service node of type Solo, Raft, and SmartBFT would be three entirely different binaries.

When first looking at the diagram I thought Solo, Raft, and SmartBFT were each consensus instances. But then I realized there would be a single consensus type per binary and the instances were per channel instances of that consensus type. This should be clarified to avoid confusion.

Reply:
> and the instances were per channel instances of that consensus type. This should be clarified to avoid confusion.

Well it makes little sense to have an orderer that participates in CFT in one channel and in BFT in another channel. Either you trust the organizations in your network or you don't. 

And if you want to connect to two different and mutually exclusive networks you can always just run another orderer. 

It's not like we have 100 different consensus implementations and you need to use each of them.